teach.md

```diff
@@ -0,0 +1,217 @@
+
+## Why Teach Scala
+
+### Multiple Paradigms
+
+While we learn new concepts we compare and contrast different but similar
+concept and relate to what we already know. This is a very important learning
+principle and many pedagogical strategies build on contrasting and comparing
+while exploring similarities and differences between concepts and ideas. Scala
+is an excellent language in supporting this pedagogical principle as it
+includes a pragmatic mix of concepts and ideas, ranging from simple to
+advanced, that can be contrasted and compared, step by step.
+
+> “One example I use in my teaching is contrasting different ways of solving the
+> configuration problem: early on we contrast a global mutable variable versus
+> default arguments and later when we get more advanced we contrast default
+> arguments with context parameters. This way we can get deep into the more
+> general discussion of the tradeoff between flexibility and complexity.”
+> 
+> Björn Regnell, Lund University, Sweden.
+
+> “We teach how to implement basic interpreters and type checkers. For this
+> purpose, algebraic data types with pattern matching and immutable data
+> structures, provided by functional languages, are essential.”
+> 
+> Jaemin Hong, KAIST, South Korea.
+
+> “Scala offers a wide variety of programming language constructs. By
+> integrating both OOP and functional aspects into one language, it is a great
+> tool to teach both paradigms and compare solutions within the same language.
+> At the same time, Scala is a great tool to teach software design: it includes
+> a whole spectrum of features ranging from basic constructs to advanced program
+> structuring techniques. This allows us to use the same language to discuss
+> various strategies to decompose a software into modules.”
+> 
+> Jonathan Brachthäuser, University of Tübingen, Germany.
+
+### Expressive
+
+Scala has a concise syntax that allows developers to focus on the intent of a
+program without being distracted by syntactic noise. Here is a “Hello world”
+program in Scala:
+
+~~~
+@main def run() = println("Hello, World!")
+~~~
+
+> “Scala is a fantastic teaching language for illustrating solutions with
+> different tradeoffs between the least complex concepts that does the job,
+> that might have lower flexibility or is less performant, compared to other
+> solutions that may be more difficult to read and maintain.”
+> 
+> Björn Regnell, Lund University, Sweden.
+
+> “Scala allows us to focus only on interesting domain concepts without dealing
+> with tedious low-level implementation issues.”
+> 
+> Jaemin Hong, KAIST, South Korea.
+
+### Safe
+
+Scala has a static type system that prevents entire classes of bugs from
+happening. The compiler helps the students to find bugs before run-time.
+If a student had too many difficult-to-hunt-down run-time bugs, this could take
+away the joy of learning and cause fatigue so badly that a student may give up
```

Comment:
```suggestion
away the joy of learning and cause fatigue such that a student may give up
```
or possibly "fatigue so bad that" (not badly) but "fatigue" is already a pretty negative word, so there's no need to add "bad" to it.

Reply:
Done in 55dfe20f